chroma_core/services/job_scheduler/job_scheduler.py

```diff
@@ -1745,3 +1748,106 @@ def update_lnet_configuration(self, lnet_configuration_list):
+                "purge_duration_active": stratagem_data.get("purge_duration_active"),
+            }
+
+            # The filesystem_id may come in as the fs name or the fs id. In terms of storing information in the database, the fs id should always be used.
+            fs_identifier = str(configuration_data.get("filesystem_id"))
+            fs_id = filter(
+                lambda x: str(x.get("id")) == fs_identifier or str(x.get("name")) == fs_identifier,
+                ManagedFilesystem.objects.values("id", "name"),
+            )[0].get("id")
+
+            configuration_data["filesystem_id"] = fs_id
+
+            with transaction.atomic():
+                matches = StratagemConfiguration.objects.filter(filesystem_id=fs_id)
+                if len(matches) == 1:
+                    matches.update(**configuration_data)
+                    stratagem_configuration = StratagemConfiguration.objects.get(filesystem_id=fs_id)
+                else:
+                    stratagem_configuration = StratagemConfiguration.objects.create(**configuration_data)
+
+            ObjectCache.add(StratagemConfiguration, stratagem_configuration)
+
+        return self.set_state(
+            [
+                (
+                    ContentType.objects.get_for_model(stratagem_configuration).natural_key(),
+                    stratagem_configuration.id,
+                    "configured",
+                )
+            ],
+            "Configuring Stratagem",
+            True,
+        )
+
+    def run_stratagem(self, mdts, stratagem_data):
+        unique_id = uuid.uuid4()
+        run_stratagem_list = map(
+            lambda mdt_id: {
+                "class_name": "RunStratagemJob",
+                "args": {
+                    "mdt_id": mdt_id,
+                    "uuid": unique_id,
+                    "report_duration": stratagem_data.get("report_duration"),
+                    "purge_duration": stratagem_data.get("purge_duration"),
+                },
+            },
+            mdts,
+        )
+
+        run_stratagem_list.append(
+            {"class_name": "AggregateStratagemResultsJob", "args": {"depends_on_job_range": range(len(mdts))}}
+        )
+
+        client_host = ManagedHost.objects.get(server_profile_id="stratagem_client")
+        client_mount_exists = LustreClientMount.objects.filter(host_id=client_host.id).exists()
+        filesystem = ManagedFilesystem.objects.get(id=ManagedMdt.objects.get(id=mdts[0]).filesystem_id)
+        mountpoint = "/mnt/{}".format(filesystem.name)
+
+        if not client_mount_exists:
+            self._create_client_mount(client_host, filesystem, mountpoint)
+
+        client_mount = LustreClientMount.objects.get(host_id=client_host.id)
+        client_mount.state = "unmounted"
+        client_mount.mountpoint = mountpoint
+        client_mount.filesystem_id = filesystem.id
+        client_mount.save()
+
+        run_stratagem_list.append(
```

Comment:
Does this need to be appended if the mount exists?

Reply:
Yes, we want it to always try to mount on each run. If it's already mounted then it will come back and simply update the state.